x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -799,6 +800,29 @@ public void testMissingInputParams() {
         expectError("row x = ?, y = ?", List.of(new TypedParamValue("integer", 1)), "Not enough actual parameters 1");
     }
 
+    public void testFieldContainingDotsAndNumbers() {
+        LogicalPlan where = processingCommand("where `a.b.1m.4321`");
+        assertThat(where, instanceOf(Filter.class));
+        Filter w = (Filter) where;
+        assertThat(w.child(), equalTo(PROCESSING_CMD_INPUT));
+        assertThat(Expressions.name(w.condition()), equalTo("a.b.1m.4321"));
+    }
+
+    public void testFieldQualifiedName() {
+        LogicalPlan where = processingCommand("where a.b.`1m`.`4321`");
+        assertThat(where, instanceOf(Filter.class));
+        Filter w = (Filter) where;
+        assertThat(w.child(), equalTo(PROCESSING_CMD_INPUT));
+        assertThat(Expressions.name(w.condition()), equalTo("a.b.1m.4321"));
+    }
+
+    public void testQuotedName() {
+        // row `my-field`=123 | stats count(`my-field`) | eval x = `count(`my-field`)`
+        LogicalPlan plan = processingCommand("stats count(`my-field`) |  keep `count(my-field)`");
```

Comment:
Good catch - thanks for this.
This was incorrect  - the correct syntax should be 
```
keep `count(``my-field``)`  ; the backquotes are escaped
```
since we want to differentiate between
```
stats count(a - b)  ; which we want to support at some point
keep `count(a-b)`  

stats count(`a-b`)   ; this is not an expression but a field name hence the ` need to be kept around meaning
| keep `count(``a``-``b``)`

; this means any backquote from the user is kept as is since we're using the query verbatim

stats count(`a`-`b`)         ; need to preserve the quotes
| keep `count(``a``-``b``)` ; 
```

I've added a CSV test and caught a bug in our identifier handling.
I've tried to play with removing redundant quotes however as a user I found the behavior surprising.

Reply:
Thanks @costin. This really LGTM.